### PR TITLE
feat(apm): Be able to click and drag over the transaction minimap to select a timeslice of the transaction

### PR DESCRIPTION
**Pull request by dashed**
Visuals:
Kapture 2019-09-19 at 19 03 16

-------

## TODO

- [ ] optimization: create the window selection using DOM, so as to not invoke the react reconciler
- [ ] add something to the time axis
- [x] better variable names; rename "custom window" to "window selection"
- [ ] reset button

-------

Closes SEN-1035

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/spans/dragManager.tsx [hunk 1]
```diff
@@ -211,12 +372,27 @@ class DragManager extends React.Component<DragManagerProps, DragManagerState> {
...
       onLeftHandleDragStart: this.onLeftHandleDragStart,
       leftHandlePosition: this.state.leftHandlePosition,
-      viewWindowStart: this.state.viewWindowStart,
+
+      // right handle
 
       onRightHandleDragStart: this.onRightHandleDragStart,
       rightHandlePosition: this.state.rightHandlePosition,
+
+      // custom window selection
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/spans/dragManager.tsx]: How is a dragged window more custom than moving sliders around? They both seem like ways to control the same window but with different control types.

**Review by markstory — APPROVED** [review]
Looks good to me from poking around in a browser and trying to make bad selections.

File: src/sentry/static/sentry/app/components/events/interfaces/spans/dragManager.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/events/interfaces/spans/dragManager.tsx]: This UX addition was one of the feedback from alpha customers; and one that didn't make the cut on the initial implementation (due to bandwidth constraints).

IMO, it's an alternative/additional option to create the selection.

**Review by markstory — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Comment by dashed** [comment]
I'm including two fixes to the transaction view:

1. fix for the improperly calculated vertical scroll offset for the minimap. this was a regression.
2. spans with equal start and end timestamps were improperly hidden if they were placed at either the beginning or end of the transaction.

**Pull request merged**